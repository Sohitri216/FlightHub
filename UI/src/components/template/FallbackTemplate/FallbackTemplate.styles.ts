import styled from "styled-components";
import { styled as Mstyled } from "@mui/system";

export const Layout = Mstyled("div")(({ theme }) => ({
  [theme.breakpoints.down("md")]: {
    width: "100%",
  },
  [theme.breakpoints.up("md")]: {
    width: "100%",
  },
}));

export const ErrorWrapper = Mstyled("div")(({ theme }) => ({
  fontFamily: "'Open Sans', sans-serif",
  margin: "1rem auto",
  padding: "0.5rem",
  boxShadow: "0.1rem 0.1rem 0.3rem #888",
  backgroundColor: "#f4f4f8",
  [theme.breakpoints.up("md")]: {
    width: "50%",
  }
}));

export const Text = styled("div")`
  padding: 1rem;
  display: flex;
  justify-content: center;
`;
