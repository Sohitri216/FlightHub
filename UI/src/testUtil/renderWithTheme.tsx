import { ThemeProvider as MThemeProvider } from "@mui/material/styles";
import { render } from "@testing-library/react";
import { Theme } from "../theme/Theme";

export const renderWithTheme = (component: JSX.Element) => {
  render(
    <MThemeProvider theme={Theme}>
      <>{component}</>
    </MThemeProvider>
  );
};
