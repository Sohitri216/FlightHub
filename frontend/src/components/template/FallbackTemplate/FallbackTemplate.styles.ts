import styled from "styled-components";

export const ErrorWrapper = styled("div")`
  font-family: "Open Sans", sans-serif;
  margin: 1rem auto;
  padding: 0.5rem;
  box-shadow: 0.1rem 0.1rem 0.3rem #888;
  width: 50%;
  background-color: #f4f4f8;
`;

export const Text = styled("div")`
  padding: 1rem;
  display: flex;
  justify-content: center;
`;
