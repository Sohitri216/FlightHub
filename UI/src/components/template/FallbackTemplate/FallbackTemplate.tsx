import * as S from "./FallbackTemplate.styles";

export const FallbackTemplate = ({ displayMessage }: Props) => {
  return (
    <S.ErrorWrapper data-testid="error-section">
      <S.Text>{displayMessage}</S.Text>
    </S.ErrorWrapper>
  );
};

type Props = {
  displayMessage: string;
};
