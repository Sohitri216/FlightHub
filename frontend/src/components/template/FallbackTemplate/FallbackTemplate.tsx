import * as S from "./FallbackTemplate.styles";

export const FallbackTemplate = ({ displayMessage }: Props) => {
  return (
    <S.ErrorWrapper>
      <S.Text>{displayMessage}</S.Text>
    </S.ErrorWrapper>
  );
};

type Props = {
  displayMessage: string;
};
