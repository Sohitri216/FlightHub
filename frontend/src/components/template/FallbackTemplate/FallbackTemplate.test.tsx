import { renderWithTheme } from "../../../testUtil/renderWithTheme";
import { FallbackTemplate } from "./FallbackTemplate";
import { screen } from "@testing-library/react";

describe("<Fallback Template /> component should", () => {
  const testProps = {
    displayMessage: "Error Message",
  };
  it("match its snapshot", () => {
    const view = renderWithTheme(<FallbackTemplate {...testProps} />);
    expect(view).toMatchSnapshot();
  });

  it("should display loading overlay sucessfully", () => {
    renderWithTheme(<FallbackTemplate {...testProps} />);
    expect(screen.getByTestId("error-section")).toBeDefined();
  });

  it("should display error message successfully", () => {
    renderWithTheme(<FallbackTemplate {...testProps} />);
    expect(screen.getByText("Error Message")).toBeDefined();
  });
});
