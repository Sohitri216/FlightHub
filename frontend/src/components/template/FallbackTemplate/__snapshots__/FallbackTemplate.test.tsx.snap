// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Fallback Template /> component should match its snapshot 1`] = `undefined`;
